wp-content/themes/sample-theme/blocks/src/core--button/block.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+namespace XWP\Sample_Theme\Blocks;
+
+use XWP\Sample_Theme\Block;
+use XWP\Sample_Theme\Block_Script;
+
+$block = new Block( 'xwp-sample-theme/button' );
+
+$block->extends( 'core/button' );
```

Comment:
This make it aware of the dependency relationship to the block that this is overriding.